### PR TITLE
[KOGITO-3382] Transitive dependency of Keycloak operator module missing

### PR DESCRIPTION
**Pull request by sutaakar**
Only keycloak-operator was updated manually, rest was updated by Go build.

Signed-off-by: Karel Suta <ksuta@redhat.com>

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Review by Sgitario — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -2,26 +2,26 @@ module github.com/kiegroup/kogito-cloud-operator
 
 require (
 	github.com/RHsyseng/operator-utils v0.0.0-20200304191317-2425bf382482
-	github.com/coreos/prometheus-operator v0.38.1-0.20200424145508-7e176fda06cc
+	github.com/coreos/prometheus-operator v0.40.0
```
Comment (ricardozanini) [inline go.mod:5]: @sutaakar could you please double check why we need this dependency? Normally people are using their `log` package meaning to use the local `log` var

**Review by ricardozanini — COMMENTED** [review]
I'm just curious about the prometheus dependency

File: go.mod (hunk 1 above)
Comment (sutaakar) [reply go.mod:5]: Actually it seems that we may not need the prometheus dependency declared. The `ServiceMonitor` seems to be created as a part of reconciliation.

**Comment by sutaakar** [comment]
CI seems down again

File: go.mod (hunk 1 above)
Comment (sutaakar) [reply go.mod:5]: So, the dependency is needed for `ServiceMonitor` declaration.
Just it seems that we can remove `PrometheusCli` from https://github.com/kiegroup/kogito-cloud-operator/blob/master/pkg/client/client.go#L62

**Comment by spolti** [comment]
@msmagnanijr  ^

File: go.mod (hunk 1 above)
Comment (sutaakar) [reply go.mod:5]: created a separate PR for PrometheusCli removal - https://github.com/kiegroup/kogito-cloud-operator/pull/579

**Review by ricardozanini — APPROVED** [review]

**Comment by sutaakar** [comment]
@ricardozanini I think this can be merged to fix operator build, functionality checked manually.

**Pull request merged**